bin/filtered_npm_audit.sh

```diff
@@ -0,0 +1,307 @@
+            if [ $_bitness = "32" ]; then
+                if [ $_ostype = "unknown-linux-gnu" ]; then
+                    # 64-bit kernel with 32-bit userland
+                    # endianness suffix is appended later
+                    local _cputype=mips
+                fi
+            else
+                # only n64 ABI is supported for now
+                local _ostype="${_ostype}abi64"
+            fi
+
+            local _cputype="$(get_endianness $_cputype "" 'el')"
+            ;;
+
+        ppc)
+            local _cputype=powerpc
+            ;;
+
+        ppc64)
+            local _cputype=powerpc64
+            ;;
+
+        ppc64le)
+            local _cputype=powerpc64le
+            ;;
+
+        *)
+            err "unknown CPU type: $_cputype"
+
+    esac
+
+    # Detect 64-bit linux with 32-bit userland
+    if [ $_ostype = unknown-linux-gnu -a $_cputype = x86_64 ]; then
+        if [ "$(get_bitness)" = "32" ]; then
+            local _cputype=i686
+        fi
+    fi
+
+    # Detect armv7 but without the CPU features Rust needs in that build,
+    # and fall back to arm.
+    # See https://github.com/rust-lang-nursery/rustup.rs/issues/587.
+    if [ $_ostype = "unknown-linux-gnueabihf" -a $_cputype = armv7 ]; then
+        if ensure grep '^Features' /proc/cpuinfo | grep -q -v neon; then
+            # At least one processor does not have NEON.
+            local _cputype=arm
+        fi
+    fi
+
+    local _arch="$_cputype-$_ostype"
+
+    RETVAL="$_arch"
+}
+
+check_ci() {
+    if [ -z ${CI+0} ]; then
+    	echo "Skipping dep lint. (set CI != 0 to run locally)";
+    	exit 0
+    fi
+}
+
+main() {
+    downloader --check
+    need_cmd uname
+    need_cmd chmod
+    need_cmd npm
+    check_ci
+
+    get_architecture || return 1
+    local _sha256
+    local _arch="$RETVAL"
+    assert_nz "$_arch" "arch"
+
+    say "Detected architecture ${_arch}"
+    if [[ $_arch == x86_64-unknown-linux* ]]; then
+	_arch=x86_64-unknown-linux-musl
+	_sha256=0f0cbb0edb582a2d6c74b44e10e7ad1abf00497a4afbe791cdca88cca3c1cbdf
+    elif [[ $_arch == i686-unknown-linux* ]]; then
+	_arch=i686-unknown-linux-musl
+	_sha256=916a41c405e05d720f90a8137e92783dc4bcff01108878980e0f32dd87160000
+    elif [[ $_arch != x86_64-apple-darwin ]]; then
+	err "Unsupported architecture $_arch"
+    else
+	# x86_64-apple-darwin
+	_sha256=be977a1937e6eb4542ac19896a3ec100c6dab3cba58b1632779f8e4b1f57ef11
+    fi
+    say "Fetching tarball for architecture ${_arch} w/ sha256: ${_sha256}"
+
+    local _tag=0.2.5
+    local _basename=audit-filter-${_tag}-${_arch}
+    local _file=${_basename}.tar.gz
+    local _url=https://github.com/mozilla-services/audit-filter/releases/download/${_tag}/${_basename}.tar.gz
+
+    cd bin/
+    test -f ${_file} || downloader "$_url" "$_file"
+    test -f ${_basename}/audit-filter || tar xvzf ${_file} ${_basename}/audit-filter > /dev/null
+    echo "${_sha256} ${_basename}/audit-filter" | sha256sum -c - > /dev/null || exit 1
```

Comment:
...oh, right. This is why, yeah? Because there's no cross-platform sha256sum binary you can depend on?

That's cool, you can ignore my other question about running in CI in that case. And I guess you can ignore my comments about POSIX-compliance and the path to bash too. If we only care about running in Travis, there's not much point jumping through hoops to run in other places!